### PR TITLE
Feature/gal/34 overview section

### PR DESCRIPTION
**Pull request by galdd**
[create overview section with child component include stats and notification, also changed the size in the css from px to rem, and fix odd/even colors

### CODE REVIEW

File: src/@types/notification.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  amount: number;
+}
+
+export interface NotificationInvitationProps {
+  avatarSrc: string;
+  listName: string;
+  responses: { accepted: boolean }[];
+}
+
+export interface InvitationNotificationProps {
```
Comment (ofir-zeitoun) [inline src/@types/notification.ts:19]: seems duplicate

File: src/@types/notification.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  avatarSrc: string;
+  listName: string;
+  responses: { accepted: boolean }[];
+}
+
+export interface NotificationExpenseProps {
```
Comment (ofir-zeitoun) [inline src/@types/notification.ts:25]: listName and avatarSrc seems to repeat, should it be another type?

File: src/assets/oz-mern.png
Comment (ofir-zeitoun) [inline src/assets/oz-mern.png:1]: BRAVO

File: src/components/dashboard/dashboard.tsx [hunk 3]
```diff
@@ -1,16 +1,12 @@
-import { ExpenseListSection } from "../expense-list-section";
-import { OverviewSection } from "../overview-section";
+import { OverviewSection } from "./overview-section";
 import "./dashboard.css";
+import { ExpenseListSection } from "./expense-list-section";
```
Comment (ofir-zeitoun) [inline src/components/dashboard/dashboard.tsx:3]: order

File: src/components/dashboard/expense-list-section/expense-list-section.css
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list-section.css:1]: COOL

File: src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.css [hunk 4]
```diff
@@ -1,5 +1,5 @@
 .expense-item.odd {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.css:1]: still exists?

File: src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.css [hunk 5]
```diff
@@ -30,30 +30,30 @@
...
-    font-size: 8px;
+    font-size: 0.5rem;
 }
 
 .expense-price {
     justify-self: center;
-    padding: 10px 20px;
-    background-color: #62b16a10;
+    padding: 0.625rem 1.25rem;
+    background-color: rgba(98, 177, 106, 0.1);
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.css:52]: why RGBA?

File: src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.tsx [hunk 6]
```diff
@@ -1,18 +1,13 @@
 import { EditOutlined } from "@ant-design/icons";
-import UserPicture from "../../../shared/user-picture/user-picture";
-import { Expense } from "../../../../@types/expense";
-import { formatDate, formatTime } from "../../../utilities/format-time";
+
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expense-item/expense-item.tsx:2]: order

File: src/components/dashboard/expense-list-section/expense-list/expenses-header/expenses-header.tsx [hunk 7]
```diff
@@ -1,8 +1,9 @@
 import { useState } from "react";
 import { DownOutlined, UpOutlined } from "@ant-design/icons";
-import { SortOrder } from "../../../../@types/sortOrderTypes";
-import { SortField } from "../../../../@types/sortField";
+
 import "./expenses-header.css";
+import { SortOrder } from "../../../../../@types/sortOrderTypes";
```
Comment (ofir-zeitoun) [inline src/components/dashboard/expense-list-section/expense-list/expenses-header/expenses-header.tsx:5]: order

File: src/components/dashboard/overview-section/notification/notification-header/notification-header.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import "./notification-header.css";
+
+export const NotificationHeader = () => {
+  return (
+    <header className="notifications-header">
+      <img
+        src="https://cdn.builder.io/api/v1/image/assets/TEMP/6ac0a0be7517c579248cdc79bc7da6e24d295c81d58f0b05e9f5a252d7127781?apiKey=9e09a71e3ebf486197b3397eee256a32&"
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/notification/notification-header/notification-header.tsx:7]: we use @ant-design/icons, no need to use another system

File: src/components/dashboard/overview-section/notification/notification.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+      expenseDescription: "Utilities",
+      listName: "Office Expenses",
+      amount: 100,
+    },
+  },
+];
+
+export const Notification = () => (
+  <div className="notifications-container">
+    {notifications.map((notification, index) => {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/notification/notification.tsx:46]: [Why Index as a key is an anti-pattern](https://medium.com/geekculture/reactjs-why-index-as-a-key-is-an-anti-pattern-4b9dc6ef0067)

File: src/components/dashboard/overview-section/notification/notification.tsx [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      if (notification.type === "expense") {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/notification/notification.tsx:47]: don't like this logic, it should be out of the component

File: src/components/dashboard/overview-section/overview-section.css [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+    display: grid;
+    grid-template-columns: repeat(2, minmax(18.75rem, 1fr));
+    gap: 1.25rem;
+    justify-content: center;
+    box-sizing: border-box;
+    width: 90%;
+}
+
+.notification-card {
+    box-shadow: 2px 2px 2px 4px rgba(201, 203, 205, 0.1);
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/overview-section.css:18]: px

File: src/components/dashboard/overview-section/overview-section.tsx [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+import { NotificationHeader } from "./notification/notification-header";
+import { useStats } from "./useStats";
+import { StatCard } from "./stat-card";
+import { Notification } from "./notification";
+import "./overview-section.css";
+
+export const OverviewSection = () => {
+  const { data: stats, isLoading, error } = useStats();
+
+  if (isLoading) {
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/overview-section.tsx:17]: make loading / error / children component

File: src/components/dashboard/overview-section/stat-card/stat-card.tsx [hunk 13]
```diff
@@ -0,0 +1,22 @@
+import { UserOutlined } from "@ant-design/icons";
+import { StatCardProps } from "../../../../@types/stats";
+import "./stat-card.css";
+
+export const StatCard = ({
+  icon = <UserOutlined />,
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/stat-card/stat-card.tsx:6]: why icon is coming from outside?

File: src/components/dashboard/overview-section/useStats.ts [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+  };
+
+  return apiFetch(`/mock/stats`, init)
+    .then((response) => {
+      if (!response.ok) {
+        throw new Error(`HTTP error! Status: ${response.status}`);
+      }
+      return response.json();
+    })
+    .then((data) => data as Stats);
```
Comment (ofir-zeitoun) [inline src/components/dashboard/overview-section/useStats.ts:29]: can be done in line 27
`return response.json() as Stats`

File: src/components/nav-bar/nav-bar.css [hunk 15]
```diff
@@ -1,31 +1,26 @@
 .navbar {
 	box-sizing: border-box;
-	padding: 0px 35px 0px 25px;
+	padding: 0 2.1875rem 0 1.5625rem;
```
Comment (ofir-zeitoun) [inline src/components/nav-bar/nav-bar.css:3]: 2.1875 is too long you can use 2.25rem (36px)
same for 1.5625, use 1.5rem (24px)

File: src/components/nav-bar/nav-bar.css [hunk 16]
```diff
@@ -34,50 +29,42 @@
 
 .user-photo span {
 	border-radius: 500px;
-	width: 46px;
-	height: 46px;
+	width: 2.875rem;
```
Comment (ofir-zeitoun) [inline src/components/nav-bar/nav-bar.css:32]: 3rem

File: src/components/nav-bar/nav-bar.css [hunk 17]
```diff
@@ -88,18 +75,18 @@
 
 .auth-container {
 	display: flex;
-	gap: 30px;
+	gap: 1.875rem;
```
Comment (ofir-zeitoun) [inline src/components/nav-bar/nav-bar.css:78]: 2rem?

File: src/components/shared/user-picture/user-picture.tsx [hunk 18]
```diff
@@ -1,6 +1,10 @@
 import { UserOutlined } from "@ant-design/icons";
 import "./user-picture.css";
-const UserPicture = ({ creatorImageUrl }: { creatorImageUrl?: string }) => {
+export const UserPicture = ({
```
Comment (ofir-zeitoun) [inline src/components/shared/user-picture/user-picture.tsx:3]: add space, use `type Props`

File: src/components/dashboard/overview-section/notification/notification.tsx (hunk 10 above)
Comment (galdd) [reply src/components/dashboard/overview-section/notification/notification.tsx:47]: Why

File: src/components/dashboard/overview-section/stat-card/stat-card.tsx (hunk 13 above)
Comment (galdd) [reply src/components/dashboard/overview-section/stat-card/stat-card.tsx:6]: I decided to allow the transfer of the Icon as a prop for flexibility and customization. This allows the component to choose which icon they want to display next to each statistic, thus making the component more versatile and dynamic.

**Review by ofir-zeitoun — APPROVED** [review]
Minor tweaks, updated

**Pull request merged**